### PR TITLE
Create and use a MapVerifier

### LINKED ISSUE #870: Map Verifier
Clients shouldn't need to reconstruct the full set of verification steps themselves. 
This logic should be packaged up, much like the log client.

### PR DESCRIPTION
**Pull request by gdbelvin**
Create a helper object for verifying responses from the Trillian Map. 

Closes #870

### CODE REVIEW

File: client/map_verifier.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+// MapVerifier verifies protos produced by the Trillian Map.
+type MapVerifier struct {
+	Hasher hashers.MapHasher
+	PubKey crypto.PublicKey
+}
+
+// NewMapVerifierFromTree creates a new MapVerifier.
+func NewMapVerifierFromTree(config *trillian.Tree) (*MapVerifier, error) {
+	if got, want := config.TreeType, trillian.TreeType_MAP; got != want {
```
Comment (phad) [inline client/map_verifier.go]: I thought this got, want style was to be restricted to test code, where it is idiomatic.
In production code it just seems to obscure the intent a little.

File: client/map_verifier.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		return nil, fmt.Errorf("Failed creating MapHasher: %v", err)
+	}
+
+	// Map Key
+	mapPubKey, err := der.UnmarshalPublicKey(config.GetPublicKey().GetDer())
+	if err != nil {
+		return nil, fmt.Errorf("Failed parsing Map public key: %v", err)
+	}
+
+	return &MapVerifier{
```
Comment (phad) [inline client/map_verifier.go]: Personally I'd make this struct initialization a one-liner as there's one two short fields being written.

File: client/map_verifier.go (hunk 1 above)
Comment (gdbelvin) [reply client/map_verifier.go]: In production code the got, want style can be helpful if the got and want values are reused inside the if block

File: client/map_verifier.go (hunk 2 above)
Comment (gdbelvin) [reply client/map_verifier.go]: done

File: client/map_verifier.go (hunk 1 above)
Comment (phad) [reply client/map_verifier.go]: ok

**Pull request merged**